### PR TITLE
Feature/code snippet

### PR DESCRIPTION
**Pull request by kevinmpowell**
Test page is located at:
`/proving-grounds/code-snippet/`

### CODE REVIEW

**Comment by lancetharper** [comment]
I think we can just gitignore the `dist` folder too. Do we really care about tracking the built site?

File: docs/.vuepress/components/cdr-doc-code-snippet.vue [hunk 1]
```diff
@@ -0,0 +1,232 @@
+<template>
+  <div class="cdr-doc-code-snippet" v-bind:class="{ 'cdr-doc-code-snippet--show-copied-notification': copied, 'cdr-doc-code-snippet--no-line-numbers': !lineNumbers,
```
[the remark this answers is not in the document]
Comment (kevinmpowell) [reply docs/.vuepress/components/cdr-doc-code-snippet.vue]: yes! I copied from Vue docs cause I'm still a newb. So there's a reason, but not a good one. I'll update that.

File: docs/.vuepress/components/cdr-doc-code-snippet.vue [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    'cdr-doc-code-snippet--no-max-height': !maxHeight }">
+    <div class="cdr-doc-code-snippet__actions" v-if="copyButton">
+      <div class="cdr-doc-code-snippet__copy-action cdr-doc-code-snippet__action" v-on:click="copyToClipBoard">
+        <cdr-button class="cdr-doc-code-snippet__copy-action">
+          <img class="cdr-doc-code-snippet__action-icon" :src="$withBase('/Copy@2x.png')" />
```
[the remark this answers is not in the document]
Comment (kevinmpowell) [reply docs/.vuepress/components/cdr-doc-code-snippet.vue]: good catch. Is `cdr-sr-only` already available in VuePress or do I need to `import` that from a package?

File: docs/.vuepress/components/cdr-doc-code-snippet.vue [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+        </cdr-button>
+        <div class="cdr-doc-code-snippet__notification">
+          <span class="cdr-doc-code-snippet__notification-message">
+            Copied!
```
[the remark this answers is not in the document]
Comment (kevinmpowell) [reply docs/.vuepress/components/cdr-doc-code-snippet.vue]: 👍

File: docs/.vuepress/components/cdr-doc-code-snippet.vue [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+          <span class="cdr-doc-code-snippet__notification-message" v-show="copyNotSupported">
+            This browser does not support automatic copying to clipboard.
+          </span>
+        </div>
+      </div>
+      <a class="cdr-doc-code-snippet__action" :href="repositoryHref" v-if="repositoryHref">
+        <img class="cdr-doc-code-snippet__action-icon" :src="$withBase('/Github@2x.png')"/>
+      </a>
+      <a class="cdr-doc-code-snippet__action" :href="sandboxHref" v-if="sandboxHref">
+        <img class="cdr-doc-code-snippet__action-icon" :src="$withBase('/CodeSandbox@2x.png')" />
```
[the remark this answers is not in the document]
Comment (kevinmpowell) [reply docs/.vuepress/components/cdr-doc-code-snippet.vue]: This was the fastest way to match the visual. I'd recommend doing an SVG refactor when we know how we'll handle "doc only" icons separate from official "cedar" icons.

File: docs/.vuepress/components/cdr-doc-code-snippet.vue [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+    triggerCopyNotSupported() { 
+      this.copied = false;
+      this.copyError = false;
+      this.copyNotSupported = true;
+    }
+  }
+}
+</script>
+<style lang="scss">
+  @import '../theme/styles/cdr-tokens.scss';
```
[the remark this answers is not in the document]
Comment (kevinmpowell) [reply docs/.vuepress/components/cdr-doc-code-snippet.vue]: I don't follow this question. Should the tokens be in `cedar.js` ? I don't know what `cedar.js` is or how scss variables would be referenced from it. Can you clarify?

File: docs/.vuepress/components/cdr-doc-code-snippet.vue [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+      </a>
+    </div>
+    <div class="cdr-doc-code-snippet__code-wrap" ref="codeWrap">
+      <div class="cdr-doc-code-snippet__code" ref="source"><slot/></div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { CdrButton } from '@rei/cdr-button';
```
[the remark this answers is not in the document]
Comment (kevinmpowell) [reply docs/.vuepress/components/cdr-doc-code-snippet.vue]: That's typically how I set up doc site environments, all components available globally, but I'm unsure how to do that auto-wiring in VuePress. It's something I can dig into if that's a priority, but I suspect someone else with more Vue experience could do it faster. Thoughts on how to proceed?

File: docs/.vuepress/components/cdr-doc-code-snippet.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (lancetharper) [reply docs/.vuepress/components/cdr-doc-code-snippet.vue]: It's in `cdr-core.css` so you should just be able to use the class

File: docs/.vuepress/components/cdr-doc-code-snippet.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (lancetharper) [reply docs/.vuepress/components/cdr-doc-code-snippet.vue]: Your use is correct here @kevinmpowell

File: docs/.vuepress/components/cdr-doc-code-snippet.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (lancetharper) [reply docs/.vuepress/components/cdr-doc-code-snippet.vue]: I was messing around with doing it yesterday, I'll commit how I was doing it and you can take a look

File: docs/.vuepress/components/cdr-doc-code-snippet.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (mhewson) [reply docs/.vuepress/components/cdr-doc-code-snippet.vue]: my bad - tech not living up to my expectations of magic

**Pull request merged**